File: Minimum Absolute Difference in BST - Leetcode 530/Minimum Absolute Difference in BST - Leetcode 530.java
class Solution {
    private TreeNode prev = null;
    private int minDiff = Integer.MAX_VALUE;

    public int getMinimumDifference(TreeNode root) {
        inOrderTraversal(root);
        return minDiff;
    }

    private void inOrderTraversal(TreeNode node) {
        if (node == null) {
            return;
        }

        // Traverse the left subtree
        inOrderTraversal(node.left);

        // Process the current node
        if (prev != null) {
            minDiff = Math.min(minDiff, node.val - prev.val);
        }

        // Update the previous node to the current node
        prev = node;

        // Traverse the right subtree
        inOrderTraversal(node.right);
    }
}

File: Minimum Absolute Difference in BST - Leetcode 530/Minimum Absolute Difference in BST - Leetcode 530.cpp
#include <climits>

class Solution {
public:
    int getMinimumDifference(TreeNode* root) {
        prev = nullptr;
        minDiff = INT_MAX;
        inOrderTraversal(root);
        return minDiff;
    }

private:
    TreeNode* prev;
    int minDiff;

    void inOrderTraversal(TreeNode* node) {
        if (node == nullptr) {
            return;
        }

        // Traverse left subtree
        inOrderTraversal(node->left);

        // Compute the minimum difference
        if (prev != nullptr) {
            minDiff = std::min(minDiff, node->val - prev->val);
        }

        // Update previous node
        prev = node;

        // Traverse right subtree
        inOrderTraversal(node->right);
    }
};
